src/syscheckd/registry/events.c

```diff
@@ -0,0 +1,284 @@
+
+cJSON *fim_registry_value_attributes_json(const fim_registry_value_data *data, const registry *configuration) {
+    cJSON *attributes = cJSON_CreateObject();
+
+    cJSON_AddStringToObject(attributes, "type", "registry_value");
+
+    if (configuration->opts & CHECK_TYPE) {
+        cJSON_AddNumberToObject(attributes, "registry_type", data->type);
+    }
+
+    if (configuration->opts & CHECK_SIZE) {
+        cJSON_AddNumberToObject(attributes, "size", data->size);
+    }
+
+    if (configuration->opts & CHECK_MD5SUM) {
+        cJSON_AddStringToObject(attributes, "hash_md5", data->hash_md5);
+    }
+
+    if (configuration->opts & CHECK_SHA1SUM) {
+        cJSON_AddStringToObject(attributes, "hash_sha1", data->hash_sha1);
+    }
+
+    if (configuration->opts & CHECK_SHA256SUM) {
+        cJSON_AddStringToObject(attributes, "hash_sha256", data->hash_sha256);
+    }
+
+    if (*data->checksum) {
+        cJSON_AddStringToObject(attributes, "checksum", data->checksum);
+    }
+
+    return attributes;
+}
+
+cJSON *fim_registry_compare_value_attrs(const fim_registry_value_data *new_data,
+                                        const fim_registry_value_data *old_data,
+                                        const registry *configuration) {
+    cJSON *changed_attributes = cJSON_CreateArray();
+
+    if ((configuration->opts & CHECK_SIZE) && old_data->size != new_data->size) {
+        cJSON_AddItemToArray(changed_attributes, cJSON_CreateString("size"));
+    }
+
+    if ((configuration->opts & CHECK_TYPE) && old_data->type != new_data->type) {
+        cJSON_AddItemToArray(changed_attributes, cJSON_CreateString("type"));
+    }
+
+    if ((configuration->opts & CHECK_MD5SUM) && (strcmp(old_data->hash_md5, new_data->hash_md5) != 0)) {
+        cJSON_AddItemToArray(changed_attributes, cJSON_CreateString("md5"));
+    }
+
+    if ((configuration->opts & CHECK_SHA1SUM) && (strcmp(old_data->hash_sha1, new_data->hash_sha1) != 0)) {
+        cJSON_AddItemToArray(changed_attributes, cJSON_CreateString("sha1"));
+    }
+
+    if ((configuration->opts & CHECK_SHA256SUM) && (strcmp(old_data->hash_sha256, new_data->hash_sha256) != 0)) {
+        cJSON_AddItemToArray(changed_attributes, cJSON_CreateString("sha256"));
+    }
+
+    return changed_attributes;
+}
+
+cJSON *fim_registry_value_json_event(const fim_entry *new_data,
+                                     const fim_entry *old_data,
+                                     const registry *configuration,
+                                     fim_event_mode mode,
+                                     unsigned int type,
+                                     __attribute__((unused)) whodata_evt *w_evt,
+                                     const char *diff) {
+    cJSON *changed_attributes;
+    char path[OS_SIZE_512];
+
+    if (old_data != NULL) {
+        changed_attributes =
+        fim_registry_compare_value_attrs(new_data->registry_entry.value, old_data->registry_entry.value, configuration);
```

Comment:
Please, move the `fim_registry_compare_value_attrs` function to the same line as the `changed_attributes` variable.
```suggestion
        changed_attributes = fim_registry_compare_value_attrs(new_data->registry_entry.value,
                                                              old_data->registry_entry.value, configuration);
```

Reply:
Done